gems/aws-sdk-core/lib/aws-sdk-core/process_credentials.rb

```diff
@@ -19,40 +25,50 @@ class ProcessCredentials
     # Creates a new ProcessCredentials object, which allows an
     # external process to be used as a credential provider.
     #
-    # @param [String] process Invocation string for process
-    # credentials provider.
+    # @param [Array<String>, String] process An array of strings including
+    #  the process name and its arguments to execute, or a single string to be
+    #  executed by the shell (deprecated and insecure).
     def initialize(process)
+      if process.is_a?(String)
```

Comment:
Ideally we would rather splat arguments like `Aws::ProcessCredentials.new('some', 'arg')` however we could not correctly warn in these cases because we already support the string case.